File: programcreek/js/01-03-isomorphic-string.js
function isIsomorphic(val1, val2) {
    if (!val1 || !val2 || typeof val1 != 'string' || typeof val2 != 'string'
            || val1.length != val2.length) {
        return 0;
    }

    var charMap = {};

    for (var i = 0, ii = val1.length; i < ii; i ++){
        var char1 = val1.charAt(i);
        var char2 = val2.charAt(i);

        if (typeof charMap[char1] == 'undefined') {
            charMap[char1] = char2;
        }
        else {
            if (charMap[char1] != char2) {
                return 0;
            }
        }
    }

    return 1;
}


console.log(isIsomorphic('add', 'egg'));
console.log(isIsomorphic('foo', 'bar'));

File: programcreek/python/01-03-isomorphic-string.py
def is_isomorphic(val1, val2) :
    # map each char from val1 to val2 and then see as they play back if they are the same order
    char_map = dict()

    if len(val1) != len(val2) or len(val1) == 0:
        return False

    for i in range(len(val1)):

        char1 = val1[i]
        char2 = val2[i]

        if char1 not in char_map:
            char_map[char1] = char2
        else:
            mapped_char = char_map[char1]
            if char2 != mapped_char:
                return False

    return True


print is_isomorphic('egg', 'add')
print is_isomorphic('foo', 'bar')